chore(deps): update jsonwebtoken to version 10

Xynnn007 opened the pull request.

Xynnn007 committed:
chore(deps): update jsonwebtoken to version 10

Signed-off-by: Xynnn007 <xynnn@linux.alibaba.com>

Xynnn007 commented:
Manual version for https://github.com/veraison/rust-ear/pull/47
ci looks like network turbulence. Probably we need a retry?

thomas-fossati commented:
> Manual version for #47 ci looks like network turbulence. Probably we need a retry?

From a quick skim, it looks like jsonwebtoken [has moved](https://github.com/Keats/jsonwebtoken/commit/c2f26101914b4b1817e732dbb74b1ed493d0592c) the Rust edition from 2021 to 2024.
Embarkstudio’s cargo-deny action uses an older version of cargo by default, which cannot handle the new Rust edition.
However, the [documentation](https://github.com/EmbarkStudios/cargo-deny-action/blob/main/README.md) suggests that it may be possible to tweak this using the `rust-version` parameter and setting it to 1.85.0.

“””maybe””” 😄

Xynnn007 committed:
ci/cargo-deny: update rust version to 1.85.0

By default the rust version that `cargo-deny-action` uses is 1.71.0,
which cannot detect dependencies with edition `2024`. Thus we manually
update it.

Signed-off-by: Xynnn007 <xynnn@linux.alibaba.com>

thomas-fossati commented:
@Xynnn007 thanks for the changes!

To make the license checks green, we need to add BSD-3-Clause to the `allow` section in deny.toml.

Xynnn007 committed:
deny.toml: add BSD-3-Clause to allowlist

Signed-off-by: Xynnn007 <xynnn@linux.alibaba.com>

thomas-fossati reviewed (approved):
🚢 it!

(Thanks for being persistent!)
